pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/MapField.java

```diff
@@ -0,0 +1,153 @@
+package com.hedera.pbj.compiler.impl;
+
+import java.util.Set;
+import com.hedera.pbj.compiler.impl.grammar.Protobuf3Parser;
+import static com.hedera.pbj.compiler.impl.SingleField.getDeprecatedOption;
+
+/**
+ * A field of type map.
+ * <p>
+ * In protobuf, a map is essentially a repeated map entry message with two fields: key and value.
+ * However, we don't model the map entry message explicitly for performance reasons. Instead,
+ * we deal with the keys and values directly, and define synthetic Field objects for them here
+ * for convenience, so that we can reuse the majority of the code generation code.
+ * <p>
+ * In model implementations we use a custom implementation of the Map interface named PbjMap
+ * which is an immutable map that exposes a SortedKeys list which allows one to iterate
+ * the map deterministically which is useful for serializing, computing the hash code, etc.
+ */
+public record MapField(
+        /** A synthetic "key" field in a map entry. */
+        Field keyField,
+        /** A synthetic "value" field in a map entry. */
+        Field valueField,
+        // The rest of the fields below simply implement the Field interface:
+        boolean repeated,
+        int fieldNumber,
+        String name,
+        FieldType type,
+        String protobufFieldType,
+        String javaFieldTypeBase,
+        String methodNameType,
+        String parseCode,
+        String javaDefault,
+        String parserFieldsSetMethodCase,
+        String comment,
+        boolean deprecated
+) implements Field {
+
+    /**
+     * Construct a MapField instance out of a MapFieldContext and a lookup helper.
+     */
+    public MapField(Protobuf3Parser.MapFieldContext mapContext, final ContextualLookupHelper lookupHelper) {
+        this(
+                new SingleField(
+                        false,
+                        FieldType.of(mapContext.keyType(), lookupHelper),
+                        1,
+                        "___" + mapContext.mapName().getText() + "__key",
+                        null,
+                        null,
+                        null,
+                        null,
+                        "An internal, private map entry key for " + mapContext.mapName().getText(),
+                        false,
+                        null),
+                new SingleField(
+                        false,
+                        FieldType.of(mapContext.type_(), lookupHelper),
+                        2,
+                        "___" + mapContext.mapName().getText() + "__value",
+                        mapContext.type_().messageType() == null ? null : mapContext.type_().messageType().messageName().getText(),
```

Comment:
Please, refactor these parameters into method like `extractMessageTypeName` to make the mode more readable.

Reply:
Good idea. I implemented them as static members in Field and refactored both the MapField and the SingleField to use them.